File: leet_easy_c++/leet_268/leet_268a.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        unordered_map<int, int> umap;
        int len=nums.size();
        for(int i=0; i<len; i++)
            umap[*(nums.begin()+i)]=1;
        for(int i=0; i<=len; i++)
            if(umap.find(i)==umap.end()) 
                return i;
    }
};

File: leet_easy_javascript/leet_268/leet_268a.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    var n = nums.length, res = 0;
    for(var i in nums){
        res ^= nums[i];
        res ^= i;
    } 
    return res^n;
};